include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort_one_wg.h

```diff
@@ -249,11 +235,41 @@ struct __subgroup_radix_sort
                                 {
                                     const uint16_t __r = __indices[__i];
                                     if (__r < __n)
-                                        __src[__r] = __values[__i];
+                                        __src[__r] = ::std::move(__values.__v[__i]);
                                 }
+                                //destroy values
+                                _ONEDPL_PRAGMA_UNROLL
+                                for (uint16_t __i = 0; __i < __block_size; ++__i)
+                                    __values.__v[__i].~_ValT();
```

Comment:
Could we ever be destroying values which were never constructed? 
(The values we avoid in `__block_load()` because of going past `__n`)